### PR TITLE
Add ef entities gas

### PR DESCRIPTION
**Pull request by LBHCallumM**

### CODE REVIEW

File: RepairsApi.Tests/V2/Controllers/GasServiceControllerTests.cs [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+
+            var responseContent = GetResultData<List<GasServiceDataResponse>>(response);
+            var statusCode = GetStatusCode(response);
+
+            statusCode.Should().Be((int) HttpStatusCode.OK);
+            responseContent.Should().BeEquivalentTo(useCaseResponse);
+        }
+
+        [Test]
+        public async Task Get_WhenCalledWithIdAndOtherSearchTerm_ReturnsBadRequest()
```
Comment (Asura5) [inline RepairsApi.Tests/V2/Controllers/GasServiceControllerTests.cs:72]: Could add "When gets called with nothing, returns a 400"

File: RepairsApi.Tests/V2/Gateways/GasServiceGatewayTests.cs [hunk 2]
```diff
@@ -0,0 +1,249 @@
...
+
+            records.Add(_fixture
+               .Build<GasServiceData>()
+               .With(x => x.Id, 9)
+               .With(x => x.Address, Address)
+               .Create());
+            return records;
+        }
+
+        private static async Task AddToDb(List<GasServiceData> records)
```
Comment (Asura5) [inline RepairsApi.Tests/V2/Gateways/GasServiceGatewayTests.cs:173]: We're going to need these in many test files over the course of gas, so might be worth separating in to a testhelper

File: RepairsApi.Tests/V2/UseCase/GetDuplicateGasServiceDataByAddressUseCaseTests.cs [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+
+            _gasServiceGatewayMock
+                .Setup(x => x.GetDuplicateAddresses())
+                .ReturnsAsync(gatewayResponse);
+
+            // Act
+            var response = await _classUnderTest.Execute();
+
+            // Assert
+            response.Should().BeEquivalentTo(gatewayResponse.ToResponse());
```
Comment (Asura5) [inline RepairsApi.Tests/V2/UseCase/GetDuplicateGasServiceDataByAddressUseCaseTests.cs:55]: Gateway should be called once and once only

File: RepairsApi/V2/Gateways/GasServiceGateway.cs [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+{
+    public class GasServiceGateway : IGasServiceGateway
+    {
+        private readonly RepairsContext _repairsContext;
+        private readonly ILogger<GasServiceGateway> _logger;
+
+        public GasServiceGateway(RepairsContext repairsContext, ILogger<GasServiceGateway> logger)
+        {
+            _repairsContext = repairsContext;
+            _logger = logger;
```
Comment (Asura5) [inline RepairsApi/V2/Gateways/GasServiceGateway.cs:22]: Doesn't look like the logger's getting much action - include some logging?

**Review by Asura5 — APPROVED** [review]

File: RepairsApi.Tests/V2/Controllers/GasServiceControllerTests.cs (hunk 1 above)
Comment (LBHCallumM) [reply RepairsApi.Tests/V2/Controllers/GasServiceControllerTests.cs:72]: Added

File: RepairsApi.Tests/V2/Gateways/GasServiceGatewayTests.cs (hunk 2 above)
Comment (LBHCallumM) [reply RepairsApi.Tests/V2/Gateways/GasServiceGatewayTests.cs]: Added

**Comment by Asura5** [comment]
Also, code smell on the use case https://sonarcloud.io/project/issues?resolved=false&types=CODE_SMELL&pullRequest=874&id=LBHackney-IT_repairs-api-dotnet&open=AYOjfqBTmmF_4EaBrQdf

**Pull request merged**